app-libs/parentchain-specs/src/lib.rs

```diff
@@ -19,49 +19,54 @@
-pub const POLKADOT_RELAY_GENESIS_HASH_HEX: &str =
-	"91b171bb158e2d3848fa23a9f1c25182fb8e20313b2c1eb49219da7a70ce90c3";
-pub const ASSET_HUB_POLKADOT_GENESIS_HASH_HEX: &str =
-	"68d56f15f85d3136970ec16946040bc1752654e906147f7e43e9d539d7c3de2f";
+pub const PASEO_RELAY_GENESIS_HASH_HEX: [u8; 32] =
+	hex!("77afd6190f1554ad45fd0d31aee62aacc33c6db0ea801129acb813f913e0764f");
+pub const ASSET_HUB_PASEO_GENESIS_HASH_HEX: [u8; 32] =
+	hex!("d6eec26135305a8ad257a20d003357284c8aa03d0bdb2b357ab0a22371e11ef2");
+pub const INTEGRITEE_PASEO_GENESIS_HASH_HEX: [u8; 32] =
+	hex!("1b69c462cd7dfea0e855c2008b66490cc8bbe90bc80b297ec0896a1c0941ce15");
+pub const INTEGRITEE_KUSAMA_GENESIS_HASH_HEX: [u8; 32] =
+	hex!("cdedc8eadbfa209d3f207bba541e57c3c58a667b05a2e1d1e86353c9000758da");
+pub const INTEGRITEE_POLKADOT_GENESIS_HASH_HEX: [u8; 32] =
+	hex!("e13e7af377c64e83f95e0d70d5e5c3c01d697a84538776c5b9bbe0e7d7b6034c");
+pub const KUSAMA_RELAY_GENESIS_HASH_HEX: [u8; 32] =
+	hex!("b0a8d493285c2df73290dfb7e61f870f17b41801197a149ca93654499ea3dafe");
+pub const ASSET_HUB_KUSAMA_GENESIS_HASH_HEX: [u8; 32] =
+	hex!("48239ef607d7928874027a43a67689209727dfb3d3dc5e5b03a39bdc2eda771a");
+pub const POLKADOT_RELAY_GENESIS_HASH_HEX: [u8; 32] =
+	hex!("91b171bb158e2d3848fa23a9f1c25182fb8e20313b2c1eb49219da7a70ce90c3");
+pub const ASSET_HUB_POLKADOT_GENESIS_HASH_HEX: [u8; 32] =
+	hex!("68d56f15f85d3136970ec16946040bc1752654e906147f7e43e9d539d7c3de2f");
 
 /// modify this for testing if necessary (brittle)
-pub const LOCAL_TEST_GENESIS_HASH_HEX: &str =
-	"6ca6d29ad6c4a200c4af356f74f03d6467dbc8a6e9ef225a2e672a990e1c7ead";
-
+pub const LOCAL_TEST_GENESIS_HASH_HEX: [u8; 32] =
+	hex!("6ca6d29ad6c4a200c4af356f74f03d6467dbc8a6e9ef225a2e672a990e1c7ead");
 /// LOCAL ASSET_HUB_ROCOCO (brittle)
-pub const ASSET_HUB_LOCAL_TEST_GENESIS_HASH_HEX: &str =
-	"af94f065f724b64ec40a7dd7ca3d25b3493d462f1e991b979e7683ae2a5da8d6";
+pub const ASSET_HUB_LOCAL_TEST_GENESIS_HASH_HEX: [u8; 32] =
+	hex!("af94f065f724b64ec40a7dd7ca3d25b3493d462f1e991b979e7683ae2a5da8d6");
 
 pub struct MinimalChainSpec {}
 
 impl MinimalChainSpec {
 	pub fn decimals(genesis_hash: Hash) -> u8 {
-		let genesis_hash_hex = hex::encode(genesis_hash);
-		match genesis_hash_hex.as_ref() {
+		match genesis_hash.into() {
 			PASEO_RELAY_GENESIS_HASH_HEX | ASSET_HUB_PASEO_GENESIS_HASH_HEX => 10,
 			POLKADOT_RELAY_GENESIS_HASH_HEX | ASSET_HUB_POLKADOT_GENESIS_HASH_HEX => 10,
 			KUSAMA_RELAY_GENESIS_HASH_HEX | ASSET_HUB_KUSAMA_GENESIS_HASH_HEX => 12,
-			INTEGRITEE_PASEO_GENESIS_HASH_HEX | INTEGRITEE_KUSAMA_GENESIS_HASH_HEX => 12,
+			INTEGRITEE_PASEO_GENESIS_HASH_HEX
+			| INTEGRITEE_KUSAMA_GENESIS_HASH_HEX
+			| INTEGRITEE_POLKADOT_GENESIS_HASH_HEX => 12,
 			LOCAL_TEST_GENESIS_HASH_HEX | ASSET_HUB_LOCAL_TEST_GENESIS_HASH_HEX => 12,
 			_ => {
 				warn!(
 					"parentchain spec for genesis {} unknown. defaulting to 12 decimals",
-					genesis_hash_hex
+					hexxer::encode(genesis_hash)
```

Comment:
You can also just use the built-in formatter to display hex for hash. Debug and Display should actually work for H256 out of the box. If it doesn't, you can use `"{:#x}"`